core/domain/topic_services.py

```
@@ -1626,6 +1627,21 @@ def _are_nodes_valid_for_publishing(
+    for node in story.story_contents.nodes:
+        for skill_id in node.acquired_skill_ids:
+            all_acquired_skill_ids.add(skill_id)
+    for skill_id in all_acquired_skill_ids:
+        question_count = (
+            question_services.get_total_question_count_for_skill_ids([skill_id])
+        )
+        if question_count < 10:
+            raise utils.ValidationError(
+                'Skill %s has only %d questions. Each skill linked to a '
+                'story node must have at least 10 questions before '
+                'publishing.' % (skill_id, question_count)
```

Ruff (0.15.18)

[warning] Use format specifiers instead of percent format

(UP031)